src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/ForkScanner.java

```diff
@@ -73,6 +93,21 @@
 
     private boolean walkingFromFinish = false;
 
+    protected NodeType currentType;
+    protected NodeType nextType;
```

Comment:
🐛 Uninitialized fields, which are being used in child classes. At least assign null